Makefile

```diff
@@ -51,7 +51,7 @@ run: use.develop
 
 .PHONY: test
 ## Run tests
-test:
+test: fmt
```

Comment:
is there a reason to require code formatting before running tests?

I think we may want to keep the targets separate, i.e. allow tests to be run independent of linting the code

Reply:
I usually do this, because at list myself, I always forget to run fmt before sending a pr.

We could then, add something like, make before-pr and do these kind of stuff on this step?

Reply:
removed, let me know if we can change it as I suggested.
If so I'll open a issue.

Reply:
yeah, I suppose we could add a Make target to run all pertinent checks in one, but `test` should be possible to run in isolation, e.g. for CI/CD

Reply:
sounds good, thanks.